packages/react/src/components/SelectUsersModal/SelectUsersModal.jsx

```diff
@@ -206,30 +217,32 @@ const SelectUsersModal = ({ isOpen, onClose, onSubmit, users, initialSelectedUse
     });
   };
 
-  const flattenUsers = useCallback((results, user) => {
-    return user.children && user.children.length > 0
-      ? user.children.reduce(flattenUsers, results)
-      : results.concat(user);
-  }, users);
-
   const usersList = displayAllUsersList(users, selectedUsers);
-  const userCount = usersList.reduce(flattenUsers, []).length;
+  const userCount = useMemo(() => usersList.reduce(flattenUsers, []).length, [usersList]);
```

Comment:
moved flattenUsers out of the component since it wasn't dependent on any state and memorized the userCount call.